### DISCUSSION TITLE
如何使用多条流同时传输？

### DISCUSSION THREAD

**Discussion by nixwl** [Tencent/tquic#188] [Q&A (问题和回答)]
您好：
我在使用 tquic 0.7  test_server、test_client 测试发现：server 侧只使用了 stream id=0 这一条流发送，如果我想改成使用多条流发送，该如何修改代码呢 ？可否提供一些建议

**Comment by glime-ai** [comment]
您好，

在 TQUIC 的 `test_server` 和 `test_client` 示例中，默认情况下可能只使用了一个 stream（流ID为0）进行数据传输。如果您想要修改代码以支持多条流发送数据，您需要在 `test_server` 的代码中创建多个 stream 并向它们发送数据。

一般来说，您可以通过以下步骤来实现这一点：

1. **创建多个流**：在 `test_server` 的处理逻辑中，您可以调用创建流的函数多次，为每个想要使用的流分配一个唯一的流ID。

2. **向流发送数据**：一旦流被创建，您就可以像之前使用流ID为0的流那样，向这些新创建的流发送数据。

3. **修改客户端以接收多个流的数据**：确保 `test_client` 能够正确处理来自服务器的多个流的数据。这可能涉及到在客户端接收数据的逻辑中添加对多个流的支持。

请注意，具体的代码修改方式将取决于 TQUIC 的具体实现细节，以及您希望如何使用这些额外的流。建议您查看 TQUIC 的文档和示例代码，以获取更多关于如何操作流的信息。

此外，如果您对 Rust 语言不是很熟悉，可能还需要学习一下 Rust 的相关知识，以便更好地进行代码修改。

由于我无法直接访问 TQUIC 的源代码，以上建议是基于一般的 QUIC 实现和编程经验提出的。如果您需要更具体的指导，建议查看 TQUIC 的官方文档或在其开源项目的 issue 区域提问。

希望这些建议对您有所帮助！

**Comment by iyangsj** [comment]
@nixwl  https://github.com/tquic-group/tquic-example-rust中的示例更简单

关于使用多个流，可以参考这里的示例：
https://github.com/Tencent/tquic/blob/develop/tools/src/bin/tquic_client.rs
https://github.com/Tencent/tquic/blob/develop/tools/src/bin/tquic_server.rs

**Reply by nixwl** [reply]
好的，我去看看
